src/app/playlist/components/TracksList/TracksList.tsx

```diff
@@ -1,19 +1,62 @@
 "use client";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./TracksList.css";
-import { playlistDataState, PlaylistData } from "@/app/recoil/atoms";
-import { useRecoilValue } from "recoil";
+import {
+  playlistDataState,
+  PlaylistData,
+  filterOptionsState,
+} from "@/app/recoil/atoms";
+import { useRecoilValue, useRecoilState } from "recoil";
 import Track from "../Track/Track";
 import { TrackObject } from "../Track/Track";
 
 const TracksList: React.FC = () => {
-  const playlistData = useRecoilValue<PlaylistData | undefined>(
-    playlistDataState
-  );
+  const [playlistData, setPlaylistData] = useRecoilState<
+    PlaylistData | undefined
+  >(playlistDataState);
+  const [filterOptions, setFilterOptions] = useRecoilState(filterOptionsState);
+  const [tracksArr, setTracksArr] = useState<TrackObject[]>([]);
```

Comment:
I think you need to make tracksArr state as a Recoil. Because you forgot to update how many tracks were found sidebar.
![Screenshot 2023-11-21 at 17 04 44](https://github.com/LorenaCapraru/song-sieve/assets/91835307/d561bbec-5254-475e-a128-d84e12ecea70)